### PR TITLE
New q audio

### PR DESCRIPTION
**Pull request by AndreyTepaykin**

### CODE REVIEW

File: platform/classes/Q/File.php [hunk 1]
```diff
@@ -59,6 +59,9 @@ static function save($params)
 
 		$tailUrl = $subpath ? "$path/$subpath/$name" : "$path/$name";
 
+		// need to send to Q/file/save after event
+		$audio = (bool)$params['audio'];
```
Comment (EGreg) [inline platform/classes/Q/File.php]: Look, `$params['audio']` may be undefined! Please fix it. !empty($params['audio']).

**Review by EGreg — CHANGES_REQUESTED** [review]
I always same thing, to [program defensively!](https://en.wikipedia.org/wiki/Defensive_programming) When writing code, think: "what if this variable is undefined" or "what if this precondition is not met"?

Another thing: why did you change the images and removed the round outlines? I was fine with the round outlines. Can you return them if you didn't have a good reason?

**Comment by AndreyTepaykin** [comment]
Fixed.
Please review again.

2017-12-21 3:13 GMT+03:00 Gregory Magarshak <notifications@github.com>:

**Comment by EGreg** [comment]
I need one more thing: please return the graphics that have circles, or explain why did you remove the circles?

**Review by EGreg — APPROVED** [review]

**Pull request merged**